packages/realm-react/src/useQuery.tsx

```diff
@@ -47,7 +47,9 @@ export function createUseQuery(useRealm: () => Realm) {
    * @param type - The object type, depicted by a string or a class extending Realm.Object
    * @returns a collection of realm objects or an empty array
    */
-  return function useQuery<T>(type: string | ({ new (): T } & Realm.ObjectClass)): Realm.Results<T & Realm.Object> {
+  return function useQuery<T>(
+    type: string | ({ new (...args: any): T } & Realm.ObjectClass),
```

Comment:
I was hoping with class models that we could simplify these types to just be `Realm.ObjectClass<T>`.  Is that possible?

Reply:
I tried but failed to update it to anything simpler. It has to match this signature: https://github.com/realm/realm-js/blob/master/types/index.d.ts#L1093